index.ts

```diff
@@ -153,13 +176,24 @@ export default class Machinomy {
    * </code></pre>
    */
   buy (options: BuyOptions): Promise<BuyResult> {
-    let _transport = transport.build()
-    let client = new Sender(this.web3, this.account, this.channelContract, _transport, this.storage, this.minimumChannelAmount, this.settlementPeriod)
-    return client.buyMeta(options).then((res: any) => {
-      return { channelId: res.payment.channelId, token: res.token }
+    const price = new BigNumber.BigNumber(options.price)
+    return this.channelManager.requireOpenChannel(this.account, options.receiver, price).then((channel: PaymentChannel) => {
+      return this.channelManager.nextPayment(channel.channelId, price, options.meta)
+        .then((payment: Payment) => this.client.doPayment(payment, options.gateway))
+        .then((res: AcceptPaymentResponse) => ({ token: res.token, channelId: id(channel.channelId) }))
     })
   }
 
+  buyUrl (uri: string): Promise<BuyResult> {
```

Comment:
Note that removing this will break the existing CLI's functionality, which is why I kept it in even though during our call we discussed taking it out.